org.eclipse.scanning.device/src/org/eclipse/scanning/device/composite/CompositeNexusScannable.java

```diff
@@ -38,13 +42,35 @@ public CompositeNexusScannable() {
 	}
 
 	@Override
-	public T getPosition() throws Exception {
+	public DeviceValueMultiPosition getPosition() throws Exception {
+		DeviceValueMultiPosition position = new DeviceValueMultiPosition();
+
+		IScannableDeviceService scannableDeviceService = getScannableDeviceService();
+		for (ChildNode childNode : getChildNodes()) {
+			final String scannableName = childNode.getScannableName();
+			try {
+				final IScannable<?> scannable = scannableDeviceService.getScannable(scannableName);
+				if (scannable instanceof CompositeNexusScannable) {
+					((CompositeNexusScannable<?>) scannable).getPosition().getValues().forEach((k,v) -> position.put(scannableName+"."+k, v));
+				} else {
+					position.put(scannableName, (double)scannable.getPosition());
+				}
+			} catch (ScanningException e) {
+				throw new NexusException(e);
+			}
+		}
+		return position;
+		/* Throwing an exception here causes the Scanning GUI to show no scannables.
 		throw new UnsupportedOperationException("A CompositeNexusScannable should only be used as a per-scan monitor");
+		return null;
+		 */
 	}
 
 	@Override
-	public T setPosition(T value, IPosition position) throws Exception {
-		throw new UnsupportedOperationException("A CompositeNexusScannable should only be used as a per-scan monitor");
+	public DeviceValueMultiPosition setPosition(DeviceValueMultiPosition value, IPosition position) throws Exception {
+		logger.warn("setPosition({}, {}) called on {}", value, position, this);
+		//throw new UnsupportedOperationException("A CompositeNexusScannable should only be used as a per-scan monitor");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)